force-app/main/default/classes/JobApplicationHandler.cls

```diff
@@ -85,4 +92,164 @@ public with sharing class JobApplicationHandler extends TriggerHandler {
+        Decimal socialSecurityCap = socialSecurityRateRecord.Maximum_Amount__c;
+        Decimal medicareRate = medicareRateRecord.Tax_Rate__c / 100;
+        Decimal medicareCap = medicareRateRecord.Maximum_Amount__c;
+        Map<String,Decimal> standardDeductionByFilingStatusMap = new Map<String,Decimal>();
+        for(Tax_Withholding__mdt deduction :standardDeductionsList){
+            if(!standardDeductionByFilingStatusMap.containsKey(deduction.Filing_Status__c)){
+                standardDeductionByFilingStatusMap.put(deduction.Filing_Status__c,deduction.Maximum_Amount__c);
+            }
+        }
+
+        //Initial map of Job Applications By Id and populate the map
+        Map<Id,Job_Application__c> jobApplictionByIdMap = new Map<Id,Job_Application__c>();
+        for(Job_Application__c jobApp :newJobApplications){
+            if(!jobApplictionByIdMap.containsKey(jobApp.Id)){
+                jobApplictionByIdMap.put(jobApp.Id, jobApp);
+            }
+        }
+
+        for(Job_Application__c jobApp :newJobApplications){
+            
+            if(jobApp.Tax_Filing_Status__c == null || jobApp.Salary__c == null || jobApp.Salary__c == 0){
+                jobApp.Federal_Income_Tax_Withheld__c = 0;
+                jobApp.Medicare_Withheld__c = 0;
+                jobApp.Social_Security_Withheld__c = 0;
+                jobApp.Take_Home_Amount__c = 0;
+                continue;
+            }
+            
+            Integer monthsInYear = 12;
+            Decimal taxableIncome = 0;
+            Decimal socialSecurityWithheld = 0;
+            Decimal medicareWithheld = 0;
+            Decimal annualSalary = annualSalaryCalculator(jobApp);
+            if(jobApp.Deduction_Type__c == 'Standard Deduction' && standardDeductionByFilingStatusMap.containsKey(jobApp.Tax_Filing_Status__c)){
+                taxableIncome = annualSalary - standardDeductionByFilingStatusMap.get(jobApp.Tax_Filing_Status__c);
+            }
+            else if(jobApp.Deduction_Type__c == 'Itemized Deduction' && jobApp.Itemized_Deduction_Amount__c != null){
+                taxableIncome = annualSalary - jobApp.Itemized_Deduction_Amount__c;
+            }
+            else{
+                taxableIncome = annualSalary;
+            }
+            if((socialSecurityCap != null && taxableIncome < socialSecurityCap) || socialSecurityCap == null){
+                socialSecurityWithheld = taxableIncome * socialSecurityRate;
+            }
+            else if(socialSecurityCap != null && taxableIncome >= socialSecurityCap){
+                socialSecurityWithheld = socialSecurityCap * socialSecurityRate;
+            }
+            if((medicareCap != null && taxableIncome < medicareCap) || medicareCap == null){
+                medicareWithheld = taxableIncome * medicareRate;
+            }
+            //Can uncomment lines 183-185 if needed. Accounts for cap on Medicare taxes, which currently don't exist
```

Comment:
Remove the reference to line numbers, these could change in the future if someone updates the code. Keep the rest of your note about being able to uncomment. Maybe consider making it a comment block to group it so no one separates it if they were to update this code.